lib/screens/v2/widget_instance/elevator_status.ex

```diff
@@ -0,0 +1,403 @@
+  end
+
+  defp serialize_closure(alert, %{name: name, id: id}, now) do
+    %{
+      elevator_name: name,
+      elevator_id: id,
+      timeframe: serialize_timeframe(alert, now),
+      description: alert.description
+    }
+  end
+
+  defp serialize_station(
+         {parent_station_id, alerts},
+         %__MODULE__{
+           facilities: facilities,
+           station_id_to_name: station_id_to_name,
+           station_id_to_icons: station_id_to_icons,
+           now: now
+         } = t
+       ) do
+    station_name = Map.fetch!(station_id_to_name, parent_station_id)
+
+    closures =
+      alerts
+      |> Enum.map(fn %Alert{
+                       informed_entities: entities
+                     } = alert ->
+        facility = get_facility_by_id(entities, facilities)
+
+        serialize_closure(alert, facility, now)
+      end)
+
+    %{
+      name: station_name,
+      icons: Map.fetch!(station_id_to_icons, parent_station_id),
+      elevator_closures: closures,
+      is_at_home_stop: parent_station_id == parent_station_id(t)
+    }
+  end
+
+  defp serialize_timeframe(%Alert{active_period: active_period} = alert, now) do
+    %{
+      happening_now: Alert.happening_now?(alert, now),
+      active_period: Enum.map(active_period, &Alert.ap_to_map/1)
+    }
+  end
+
+  defp alerts_by_station(alerts) do
+    alerts
+    |> Enum.map(fn %{informed_entities: informed_entities} = alert ->
+      {get_parent_station_id_from_informed_entities(informed_entities), alert}
+    end)
+    |> Enum.group_by(&elem(&1, 0), &elem(&1, 1))
+
+    # produces %{parent_station_id => [%Alert{}, ...]}
+  end
+
+  defp get_parent_station_id_from_informed_entities(entities) do
+    entities
+    |> Enum.find_value(fn
+      %{stop: "place-" <> _ = parent_station_id} -> parent_station_id
+      _ -> false
+    end)
+  end
+
+  defp serialize_detail_page(
+         %Alert{header: header, informed_entities: entities} = alert,
+         %__MODULE__{
+           station_id_to_icons: station_id_to_icons,
+           facilities: facilities,
+           now: now
+         } = t
+       ) do
+    facility = get_facility_by_id(entities, facilities)
+
+    %DetailPage{
+      header_text: header,
+      icons: Map.fetch!(station_id_to_icons, parent_station_id(t)),
+      elevator_closure: serialize_closure(alert, facility, now)
+    }
+  end
+
+  defp serialize_list_page(
+         alerts,
+         %__MODULE__{} = t
+       ) do
+    stations =
+      alerts
+      |> alerts_by_station()
+      |> Enum.map(&serialize_station(&1, t))
+
+    %ListPage{
+      stations: stations
+    }
+  end
+
+  def priority(_instance), do: [2]
+
+  @spec serialize(t()) :: %{pages: list(DetailPage.t() | ListPage.t())}
+  def serialize(%__MODULE__{} = t) do
+    active_at_home =
+      t
+      |> get_active_at_home_station()
+      |> Enum.map(&serialize_detail_page(&1, t))
+
+    active_elsewhere =
+      t
+      |> get_active_elsewhere()
+      |> serialize_list_page(t)
```

Comment:
I think in the final code we will also include closures at this station in the list view (it shows _all_ active closures in the designs) but that can be done without too much trouble later on.

E.g. Tufts in the list view example below is the screen's "home station"
![image](https://user-images.githubusercontent.com/63608771/152040509-09578b52-4d6c-4399-99c8-013bf5908020.png)

Reply:
Good catch. I went ahead and added it now.